### DISCUSSION TITLE
Downloading OpenVPN config bundle openvpn into temporary file ... curl: (28) Resolving timed out after 1227578879263 milliseconds

### DISCUSSION THREAD

**Discussion by hrenaud** [haugene/docker-transmission-openvpn#2077] [Q&A] [answered]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Config used

I'm on linux/arm/v7 (MyCloudEX2Ultra)

{
    "AppArmorProfile": "",
    "Args": [
        "/etc/openvpn/start.sh"
    ],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "dumb-init",
            "/etc/openvpn/start.sh"
        ],
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "OPENVPN_PROVIDER=PIA",
            "OPENVPN_CONFIG=france,sweden,austria,italy,belgium",
            "OPENVPN_USERNAME=XXXXXXX",
            "OPENVPN_PASSWORD=XXXXXXX",
            "LOCAL_NETWORK=172.17.0.0/16",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "GLOBAL_APPLY_PERMISSIONS=true",
            "TRANSMISSION_HOME=/data/transmission-home",
            "TRANSMISSION_RPC_PORT=9091",
            "TRANSMISSION_DOWNLOAD_DIR=/data/completed",
            "TRANSMISSION_INCOMPLETE_DIR=/data/incomplete",
            "TRANSMISSION_WATCH_DIR=/data/watch",
            "CREATE_TUN_DEVICE=true",
            "ENABLE_UFW=false",
            "UFW_ALLOW_GW_NET=false",
            "UFW_EXTRA_PORTS=",
            "UFW_DISABLE_IPTABLES_REJECT=false",
            "PUID=",
            "PGID=",
            "PEER_DNS=true",
            "PEER_DNS_PIN_ROUTES=true",
            "DROP_DEFAULT_ROUTE=",
            "WEBPROXY_ENABLED=false",
            "WEBPROXY_PORT=8118",
            "WEBPROXY_USERNAME=",
            "WEBPROXY_PASSWORD=",
            "LOG_TO_STDOUT=false",
            "HEALTH_CHECK_HOST=google.com",
            "SELFHEAL=false",
            "REVISION=d1ece1c5f36f275d5d950bd2557ec43f79a80d14",
            "OVERRIDE_DNS_1=1.1.1.1",
            "OVERRIDE_DNS_2=1.0.0.1",
            "DISABLE_PORT_UPDATER=true"
        ],
        "ExposedPorts": {
            "8118/tcp": {},
            "9091/tcp": {}
        },
        "Healthcheck": {
            "Interval": 60000000000,
            "Test": [
                "CMD-SHELL",
                "/etc/scripts/healthcheck.sh"
            ]
        },
        "Hostname": "85ed774cfb22",
        "Image": "haugene/transmission-openvpn:latest",
        "Labels": {
            "autoheal": "true",
            "org.opencontainers.image.revision": "d1ece1c5f36f275d5d950bd2557ec43f79a80d14",
            "org.opencontainers.image.source": "https://github.com/haugene/docker-transmission-openvpn"
        },
        "OnBuild": null,
        "OpenStdin": false,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": {
            "/config": {},
            "/data": {}
        },
        "WorkingDir": ""
    },
    "Created": "2021-11-25T02:20:38.735333364Z",
    "Driver": "vfs",
    "ExecIDs": null,
    "GraphDriver": {
        "Data": null,
        "Name": "vfs"
    },
    "HostConfig": {
        "AutoRemove": false,
        "Binds": [
            "0564846541d76ee0b6b0e372a7a8bde5a0888f719ca752418d04816efb9ff488:/data",
            "8f2d31021f3a8cb8477d9f670f71c72fdf9eea883cb1406aafc7d0b8526bcaa4:/config"
        ],
        "BlkioDeviceReadBps": null,
        "BlkioDeviceReadIOps": null,
        "BlkioDeviceWriteBps": null,
        "BlkioDeviceWriteIOps": null,
        "BlkioWeight": 0,
        "BlkioWeightDevice": null,
        "CapAdd": [
            "AUDIT_WRITE",
            "CHOWN",
            "DAC_OVERRIDE",
            "FOWNER",
            "FSETID",
            "KILL",
            "MKNOD",
            "NET_ADMIN",
            "NET_BIND_SERVICE",
            "NET_RAW",
            "SETFCAP",
            "SETGID",
            "SETPCAP",
            "SETUID",
            "SYS_CHROOT"
        ],
        "CapDrop": [
            "AUDIT_CONTROL",
            "BLOCK_SUSPEND",
            "DAC_READ_SEARCH",
            "IPC_LOCK",
            "IPC_OWNER",
            "LEASE",
            "LINUX_IMMUTABLE",
            "MAC_ADMIN",
            "MAC_OVERRIDE",
            "NET_BROADCAST",
            "SYSLOG",
            "SYS_ADMIN",
            "SYS_BOOT",
            "SYS_MODULE",
            "SYS_NICE",
            "SYS_PACCT",
            "SYS_PTRACE",
            "SYS_RAWIO",
            "SYS_RESOURCE",
            "SYS_TIME",
            "SYS_TTY_CONFIG",
            "WAKE_ALARM"
        ],
        "Capabilities": null,
        "Cgroup": "",
        "CgroupParent": "",
        "ConsoleSize": [
            0,
            0
        ],
        "ContainerIDFile": "",
        "CpuCount": 0,
        "CpuPercent": 0,
        "CpuPeriod": 0,
        "CpuQuota": 0,
        "CpuRealtimePeriod": 0,
        "CpuRealtimeRuntime": 0,
        "CpuShares": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "DeviceCgroupRules": null,
        "DeviceRequests": null,
        "Devices": [],
        "Dns": [],
        "DnsOptions": null,
        "DnsSearch": null,
        "ExtraHosts": [],
        "GroupAdd": null,
        "IOMaximumBandwidth": 0,
        "IOMaximumIOps": 0,
        "Init": false,
        "IpcMode": "private",
        "Isolation": "",
        "KernelMemory": 0,
        "KernelMemoryTCP": 0,
        "Links": null,
        "LogConfig": {
            "Config": {},
            "Type": "json-file"
        },
        "MaskedPaths": [
            "/proc/asound",
            "/proc/acpi",
            "/proc/kcore",
            "/proc/keys",
            "/proc/latency_stats",
            "/proc/timer_list",
            "/proc/timer_stats",
            "/proc/sched_debug",
            "/proc/scsi",
            "/sys/firmware"
        ],
        "Memory": 0,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "MemorySwappiness": null,
        "NanoCpus": 0,
        "NetworkMode": "bridge",
        "OomKillDisable": false,
        "OomScoreAdj": 0,
        "PidMode": "",
        "PidsLimit": null,
        "PortBindings": {
            "9091/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "9091"
                }
            ]
        },
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyPaths": [
            "/proc/bus",
            "/proc/fs",
            "/proc/irq",
            "/proc/sys",
            "/proc/sysrq-trigger"
        ],
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": "no"
        },
        "Runtime": "runc",
        "SecurityOpt": null,
        "ShmSize": 67108864,
        "UTSMode": "",
        "Ulimits": null,
        "UsernsMode": "",
        "VolumeDriver": "",
        "VolumesFrom": null
    },
    "HostnamePath": "/mnt/HD/HD_a2/Nas_Prog/_docker/containers/a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c/hostname",
    "HostsPath": "/mnt/HD/HD_a2/Nas_Prog/_docker/containers/a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c/hosts",
    "Id": "a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c",
    "Image": "sha256:4accca49b8718b7f3a2f7411c38af639e715d3aca8ef94e15c9b7e9c9e4b5218",
    "LogPath": "/mnt/HD/HD_a2/Nas_Prog/_docker/containers/a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c/a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c-json.log",
    "MountLabel": "",
    "Mounts": [
        {
            "Destination": "/data",
            "Driver": "local",
            "Mode": "z",
            "Name": "0564846541d76ee0b6b0e372a7a8bde5a0888f719ca752418d04816efb9ff488",
            "Propagation": "",
            "RW": true,
            "Source": "/mnt/HD/HD_a2/Nas_Prog/_docker/volumes/0564846541d76ee0b6b0e372a7a8bde5a0888f719ca752418d04816efb9ff488/_data",
            "Type": "volume"
        },
        {
            "Destination": "/config",
            "Driver": "local",
            "Mode": "z",
            "Name": "8f2d31021f3a8cb8477d9f670f71c72fdf9eea883cb1406aafc7d0b8526bcaa4",
            "Propagation": "",
            "RW": true,
            "Source": "/mnt/HD/HD_a2/Nas_Prog/_docker/volumes/8f2d31021f3a8cb8477d9f670f71c72fdf9eea883cb1406aafc7d0b8526bcaa4/_data",
            "Type": "volume"
        }
    ],
    "Name": "/transmission-openvpn",
    "NetworkSettings": {
        "Bridge": "",
        "EndpointID": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "HairpinMode": false,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "MacAddress": "",
        "Networks": {
            "bridge": {
                "Aliases": null,
                "DriverOpts": null,
                "EndpointID": "",
                "Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "IPAMConfig": {},
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "Links": null,
                "MacAddress": "",
                "NetworkID": "172de9bf9857b294a0218ea53afb862f2e7305573739a2d6e55109f2148d3d5f"
            }
        },
        "Ports": {},
        "SandboxID": "4d15833277a698b6ac21cde48487d583e9d3c991167cb9d8a54da55b9d883ffd",
        "SandboxKey": "/var/run/docker/netns/4d15833277a6",
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null
    },
    "Path": "dumb-init",
    "Platform": "linux",
    "Portainer": {
        "ResourceControl": {
            "Id": 90,
            "ResourceId": "a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c",
            "SubResourceIds": [],
            "Type": 1,
            "UserAccesses": [],
            "TeamAccesses": [],
            "Public": false,
            "AdministratorsOnly": true,
            "System": false
        }
    },
    "ProcessLabel": "",
    "ResolvConfPath": "/mnt/HD/HD_a2/Nas_Prog/_docker/containers/a7f5d76b7dd480edf1aa53e988cf60ef0f4418dc75a1da1d92b04f115aa9ae3c/resolv.conf",
    "RestartCount": 0,
    "State": {
        "Dead": false,
        "Error": "",
        "ExitCode": 28,
        "FinishedAt": "2021-11-25T02:21:29.780372901Z",
        "Health": {
            "FailingStreak": 0,
            "Log": [],
            "Status": "unhealthy"
        },
        "OOMKilled": false,
        "Paused": false,
        "Pid": 0,
        "Restarting": false,
        "Running": false,
        "StartedAt": "2021-11-25T02:21:27.036309811Z",
        "Status": "exited"
    }
}

### Current Behavior

When I launch the container, it doesn't seem to connect to the internet, even though I filled in the dns (Google or Cloudflare)...

### Expected Behavior

Download the OPENVPN configuration from PIA and start the rest of the boot process

### How have you tried to solve the problem?

I went through all issues talking about network, dns, curl, ARM...
I read the doc, I did step by step debug.md (I always get stuck in the same place)...

### Log output

Starting container with revision: d1ece1c5f36f275d5d950bd2557ec43f79a80d14,
One or more OVERRIDE_DNS addresses found. Will use them to overwrite /etc/resolv.conf,
/etc/openvpn/start.sh: line 32:    14 Aborted                 nslookup ${HEALTH_CHECK_HOST:-"google.com"} > /dev/null 2>&1,
WARNING: initial DNS resolution test failed,
Creating TUN device /dev/net/tun,
Using OpenVPN provider: PIA,
Running with VPN_CONFIG_SOURCE auto,
Provider PIA has a bundled setup script. Defaulting to internal config,
Executing setup script for PIA,
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.LcCn3eKwXJ,
curl: (28) Timeout was reached

### Environment

```markdown
- OS: CloudOS5 / BusyBox v1.30.1 (2020-09-04 02:41:28 UTC) built-in shell (ash)
- Docker version 19.03.8, build afacb8b
- Portainer 2.9.2
```

### Anything else?

- https://github.com/WDCommunity/wdpksrc/issues/57
- [docker==21.04.06](https://cloudsmith.io/~wd-community/repos/EX2Ultra/packages/detail/raw/docker/21.04.06/)

**Comment by pkishino** [comment]
check the lines 3+4 in the log..initial dns fails.. check your /etc/resolv.conf file in the container and try to ping from the container..looks like your network isnt working properly.

**Comment by hatchcanon** [comment]
I have a feeling this is because the port forwarding script breaks in version 4.0. If you revert back to 3.7.1 it works great.

**Comment by hrenaud** [comment]
@pkishino i can't make a ping, the container shutdown too fast. Here my resolv.conf
```
nameserver 1.1.1.1
nameserver 1.0.0.1
```
@hatchcanon 
I tried the 3.7.1 version, my logs changed, but it's still not working:
```
Starting container with revision: d1ece1c5f36f275d5d950bd2557ec43f79a80d14,
One or more OVERRIDE_DNS addresses found. Will use them to overwrite /etc/resolv.conf,
WARNING: initial DNS resolution test failed,
Creating TUN device /dev/net/tun,
Using OpenVPN provider: PIA,
Provider PIA has a custom setup script, executing it,
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.acFiOe,
curl: (35) error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time
```

**Comment by pkishino** [comment]
Ah, based on the last part, I suspect you might have the same issue as
those running  raspberry pies perhaps, needs some 3pp library update. Don’t
have the link here but it’s described in other issues if you search for
raspberry pie

**Comment by hrenaud** [comment]
I duplicated my configuration and I changed the command to `nslookup www.google.fr` and I have this error, I don't know if it could help...
```
nslookup: clock_gettime(MONOTONIC) failed
```
I'm a new user with Docker and Portainer, I'm not sure how to configure networks and which use (bridge, host, an other...)

**Comment by pkishino** [comment, marked as answer]
Run this in your host
https://github.com/haugene/docker-transmission-openvpn/issues/1726#issuecomment-865199344

this is a problem with your platform, not this container

**Comment by hrenaud** [comment]
I can get `libseccomp-dev_2.5.1-1~bpo10+1_armhf.deb` (the latest version) by wget, but `sudo dpkg -i` does not work
```
dpkg: can't extract control file
```

My NAS is : "MyCloudEX2Ultra" Firmware Version: 5.18.117.1022.2021 based on Debian Buster

**Reply by pkishino** [reply]
Yeah, I suggest googling how to upgrade libseccomp on that platform and that error etc. our checking with support. Nothing on this end to do

**Comment by hrenaud** [comment]
I just managed to make the image work.
The problem may be with my NAS, but to fix it, the solution is to add libseccomp to the image, not the server/NAS.
It would be nice to include the library directly to the image.

I changed the command to sleep 1000, so I could run it directly in the image before it goes to error

```shell
wget http://ftp.us.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.5.3-2_armhf.deb
sudo dpkg -i libseccomp2_2.5.3-2_armhf.deb
```

In Portainer:

- RUNTIME AND RESSOURCES > SET Privileged mode
- CAPABILITIES > NET ADMIN

I put back the initial command

Et voilà 🎉

**Reply by pkishino** [reply]
this won't work for other platforms though so it's not an option adding this to the image
